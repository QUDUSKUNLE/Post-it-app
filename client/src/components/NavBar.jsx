import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @export
 * @description - renders NavBar Component
 * @class NavBar
 * @extends {React.Component}
 */
export default class NavBar extends React.Component {
  render() {
    return (
      <div>
        <nav className="navbar navbar-inverse navabar-fixed-top"
          role="navigation">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle collapsed"
                data-toggle="collapse" data-target=".navbar-collapse">
                <span className="sr-only">Toggle navigation</span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
              </button>
              <Link className="navbar-brand" to="/">
                PostIt<small>App</small>
              </Link>
            </div>
            <div className="collapse navbar-collapse">
              <ul className="nav navbar-nav">
              </ul>
              <ul className="nav navbar-nav navbar-right">
                <li className="active"><Link to="/">Home</Link></li>
                <li><Link to="/signin">Sign in</Link></li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}
