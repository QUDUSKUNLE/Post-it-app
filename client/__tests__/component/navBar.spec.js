import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Link } from 'react-router';
import renderer from 'react-test-renderer';
import NavBar from '../../src/components/NavBar.jsx';

describe('PostIt-app', () => {
  it('expects NavBar component to be defined', () => {
    expect(NavBar).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<NavBar/>);
    expect(component).toMatchSnapshot();
    expect(component.find('nav').length).toBe(1);
    expect(component.find('nav')).toHaveLength(1);
    expect(component.find('div')).toHaveLength(4);
    expect(component.find('span')).toHaveLength(4);
    expect(component.find('Link')).toHaveLength(3);
    expect(component.find('ul')).toHaveLength(2);
    expect(component.find('li')).toHaveLength(2);
  });
});
