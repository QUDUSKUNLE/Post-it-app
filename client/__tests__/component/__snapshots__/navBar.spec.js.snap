// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostIt-app should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <nav
      className="navbar navbar-inverse navabar-fixed-top"
      role="navigation"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <button
            className="navbar-toggle collapsed"
            data-target=".navbar-collapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
          <Link
            className="navbar-brand"
            replace={false}
            to="/"
          >
            PostIt
            <small>
              App
            </small>
          </Link>
        </div>
        <div
          className="collapse navbar-collapse"
        >
          <ul
            className="nav navbar-nav"
          />
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li
              className="active"
            >
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/signin"
              >
                Sign in
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  "nodes": Array [
    <div>
      <nav
        className="navbar navbar-inverse navabar-fixed-top"
        role="navigation"
      >
        <div
          className="container"
        >
          <div
            className="navbar-header"
          >
            <button
              className="navbar-toggle collapsed"
              data-target=".navbar-collapse"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="sr-only"
              >
                Toggle navigation
              </span>
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
            </button>
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              PostIt
              <small>
                App
              </small>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
          >
            <ul
              className="nav navbar-nav"
            />
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li
                className="active"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign in
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NavBar />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NavBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="/"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav"
                />
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="/"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav"
                />
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NavBar />,
}
`;
